incremental-font-transfer/src/ift_graph.rs

```diff
@@ -0,0 +1,261 @@
+        //horizontal
+        Tag::new(b"calt"),
+        Tag::new(b"clig"),
+        Tag::new(b"curs"),
+        Tag::new(b"kern"),
+        Tag::new(b"rclt"),
+        //vertical
+        Tag::new(b"valt"),
+        Tag::new(b"vert"),
+        Tag::new(b"vkrn"),
+        Tag::new(b"vpal"),
+        Tag::new(b"vrt2"),
+        //ltr
+        Tag::new(b"ltra"),
+        Tag::new(b"ltrm"),
+        //rtl
+        Tag::new(b"rtla"),
+        Tag::new(b"rtlm"),
+        //random
+        Tag::new(b"rand"),
+        //justify
+        Tag::new(b"jalt"), // HarfBuzz doesn't use; others might
+        //East Asian spacing
+        Tag::new(b"chws"),
+        Tag::new(b"vchw"),
+        Tag::new(b"halt"),
+        Tag::new(b"vhal"),
+        //private
+        Tag::new(b"Harf"),
+        Tag::new(b"HARF"),
+        Tag::new(b"Buzz"),
+        Tag::new(b"BUZZ"),
+        //shapers
+
+        //arabic
+        Tag::new(b"init"),
+        Tag::new(b"medi"),
+        Tag::new(b"fina"),
+        Tag::new(b"isol"),
+        Tag::new(b"med2"),
+        Tag::new(b"fin2"),
+        Tag::new(b"fin3"),
+        Tag::new(b"cswh"),
+        Tag::new(b"mset"),
+        Tag::new(b"stch"),
+        //hangul
+        Tag::new(b"ljmo"),
+        Tag::new(b"vjmo"),
+        Tag::new(b"tjmo"),
+        //tibetan
+        Tag::new(b"abvs"),
+        Tag::new(b"blws"),
+        Tag::new(b"abvm"),
+        Tag::new(b"blwm"),
+        //indic
+        Tag::new(b"nukt"),
+        Tag::new(b"akhn"),
+        Tag::new(b"rphf"),
+        Tag::new(b"rkrf"),
+        Tag::new(b"pref"),
+        Tag::new(b"blwf"),
+        Tag::new(b"half"),
+        Tag::new(b"abvf"),
+        Tag::new(b"pstf"),
+        Tag::new(b"cfar"),
+        Tag::new(b"vatu"),
+        Tag::new(b"cjct"),
+        Tag::new(b"init"),
+        Tag::new(b"pres"),
+        Tag::new(b"abvs"),
+        Tag::new(b"blws"),
+        Tag::new(b"psts"),
+        Tag::new(b"haln"),
+        Tag::new(b"dist"),
+        Tag::new(b"abvm"),
+        Tag::new(b"blwm"),
+    ])
+}
+
+fn get_feature_tags(font: &FontRef<'_>) -> Result<BTreeSet<Tag>, ReadError> {
+    let standard_features = standard_features();
+
+    let mut result: BTreeSet<Tag> = Default::default();
+
+    if let Ok(gsub) = font.gsub() {
+        for fr in gsub.feature_list()?.feature_records() {
+            if standard_features.contains(&fr.feature_tag()) {
+                continue;
+            }
+
+            result.insert(fr.feature_tag());
+        }
+    }
+
+    if let Ok(gpos) = font.gpos() {
+        for fr in gpos.feature_list()?.feature_records() {
+            if standard_features.contains(&fr.feature_tag()) {
+                continue;
+            }
+
+            result.insert(fr.feature_tag());
+        }
+    }
+
+    Ok(result)
+}
+
+fn get_node_name(font: &FontRef<'_>) -> Result<String, ReadError> {
+    let chars: BTreeSet<char> = font
+        .charmap()
+        .mappings()
+        .map(|(cp, _)| char::from_u32(cp).unwrap())
```

Comment:
do we care if the font has invalid data here?

Reply:
No, this utility is really only meant for use in testing. Just panicking and failing in the face of a bad font is fine.